lib/mirage_impl_misc.ml

```diff
@@ -57,6 +57,22 @@ let pkg_config pkgs args =
   Bos.OS.Cmd.(run_out cmd |> out_string |> success) >>| fun data ->
   String.cuts ~sep:" " ~empty:false data
 
+(* Invoke solo5-config for target and return output if successful. *)
+let solo5_config target args =
+  let target_string = match target with
+    | `Virtio -> "--target=virtio"
+    | `Muen -> "--target=muen"
+    | `Hvt -> "--target=hvt"
+    | `Genode -> "--target=genode"
+    | `Spt -> "--target=spt"
+    | `Xen | `Qubes -> "--target=xen"
+    | _ ->
+      invalid_arg "solo5_config only defined for solo5 targets"
+  in
+  let cmd = Bos.Cmd.(v "solo5-config" % target_string %% of_list args) in
+  Bos.OS.Cmd.(run_out cmd |> out_string |> success) >>| fun data ->
+  String.cuts ~sep:" " ~empty:false data
```

Comment:
if we drop the last line `String.cuts ...` here (together with the `>>| fun data ->` in the previous line, the result of `solo5_config` will be a `string` (instead of a `string list`).